internal/own/codeowners/parse.go

```diff
@@ -0,0 +1,179 @@
+	// in such a way that for syntactic purposes, every line can be considered
+	// in isolation.
+	line string
+	// The most recently defined section, or "" if none.
+	section string
+}
+
+// nextLine advances parsing to focus on the next line.
+// Conveniently returns the same object for chaining with `notBlank()`.
+func (p *parsing) nextLine(line string) {
+	p.line = line
+}
+
+// rulePattern is expected to match a rule line like:
+// `cmd/**/docs/index.md @readme-owners owner@example.com`.
+//
+//	^^^^^^^^^^^^^^^^^^^^ ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+//
+// The first capturing   The second capturing group
+// group extracts        extracts all the owners
+// the file pattern.     separated by whitespace.
+//
+// The first capturing group supports escaping a whitespace with a `\`,
+// so that the space is not treated as a separator between the pattern
+// and owners.
+var rulePattern = regexp.MustCompile(`^\s*((?:\\.|\S)+)((?:\s+\S+)*)\s*$`)
+
+// matchRule tries to extract a codeowners rule from the current line
+// and return the file pattern and one or more owners.
+// Match is indicated by the third return value being true.
+//
+// Note: Need to check if a line matches a section using `matchSection`
+// before matching a rule with this method, as `matchRule` will actually
+// match a section line. This is because `matchRule` does not verify
+// whether a pattern is a valid pattern. A line like "[documentation]"
+// would be considered a pattern without owners (which is supported).
+func (p *parsing) matchRule() (string, []string, bool) {
+	match := rulePattern.FindStringSubmatch(p.lineWithoutComments())
+	if len(match) != 3 {
+		return "", nil, false
+	}
+	filePattern := match[1]
+	owners := strings.Fields(match[2])
+	return filePattern, owners, true
+}
+
+var sectionPattern = regexp.MustCompile(`^\s*\[([^\]]+)\]\s*$`)
+
+// matchSection tries to extract a section which looks like `[section name]`.
+func (p *parsing) matchSection() bool {
+	match := sectionPattern.FindStringSubmatch(p.lineWithoutComments())
+	if len(match) != 2 {
+		return false
+	}
+	p.section = match[1]
+	return true
+}
+
+// isBlank returns true if the current line has no semantically relevant
+// content. It can be blank while containing comments or whitespace.
+func (p *parsing) isBlank() bool {
+	return strings.TrimSpace(p.lineWithoutComments()) == ""
+}
+
+const (
+	commentStart    = rune('#')
+	escapeCharacter = rune('\\')
+)
+
+// lineWithoutComments returns the current line with the commented part
+// stripped out.
+func (p *parsing) lineWithoutComments() string {
+	// A sensible default for index of the first byte where line-comment
+	// starts is the line lenght. When the comment is removed by slcing
+	// the string at the end, using the line-length as the index
+	// of the first character dropped, yields the original string.
+	commentStartIndex := len(p.line)
+	var esc bool // whether current character is escaped.
```

Comment:
we can change variable name and remove the comment :)
```suggestion
	var isEscaped bool
```